ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -896,20 +1037,55 @@ export const RuntimePanel = fp.flow(
   if (currentRuntime === undefined) {
     return <Spinner style={{width: '100%', marginTop: '5rem'}}/>;
   }
+  const createStandardComputeRuntimeRequest = (runtime: RuntimeConfig) => {
+    // The logic here is tricky to be compatible for old runtime without PD
+    if (!pdExists || pdExists && pdReduced) {
+      return {
+        gceWithPdConfig: {
+          bootDiskSize: 50,
+          machineType: runtime.machine.name,
+          persistentDisk: {
+            name: '',
+            size: runtime.diskSize,
+            diskType: DiskType.Standard,
+            labels: {}
 
+          }
+        }
+      };
+    } else {
+      if (!runtimeExists) {
+        return {
+          gceWithPdConfig: {
+            bootDiskSize: 50,
+            machineType: runtime.machine.name,
+            persistentDisk: {
+              name: persistentDisk.name,
+              size: runtime.diskSize,
+              diskType: DiskType.Standard,
+              labels: {}
+
+            }
+          }
+        };
+      } else {
+        return {
+          gceConfig: {
+            machineType: runtime.machine.name,
+            diskSize: runtime.diskSize
+          }
+        };
+      }
+    }
+  };
```

Comment:
The whole thing can be simplified to:
```
    if(runtimeExists && !pdExists) {
      return {
        gceConfig: {
          machineType: runtime.machine.name,
          diskSize: runtime.diskSize
        }
      };
    } else {
      return {
        gceWithPdConfig: {
          bootDiskSize: 50,
          machineType: runtime.machine.name,
          persistentDisk: {
            name: pdReduced ? '' : persistentDisk.name,
            size: runtime.diskSize,
            diskType: DiskType.Standard,
            labels: {}

          }
        }
      };
    }
```

Reply:
That won't work when increasing PD size of an existing runtime, since Leo requires passing the gceConfig in this case.

Reply:
Thanks for the reference. 
Context to CH: Leo's updateRunteime
https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/swagger/api-docs.yaml#L2464
does not have gceWithPd. So to increase an attached PD size, we will use
```
  "runtimeConfig": {
    "cloudService": "gce",
    "machineType": "n1-highmem-16",
    "diskSize": 500
  }
```

Reply:
This logic is till complicated and hard to understand. Few ideas:
Create a state enum like PD_INCREASE, PD_DECREATE, NEW_RUNTIME_WITHPD. We can also use switch-case, instead of many if-else. But I am not sure what is the best way to define those states, since there are multiple ways of doing that.
One more suggestion, would be helpful to add more comments to explain. 
Would this code cover all scenarios: 
```
\\  const createStandardComputeRuntimeRequest = (runtime: RuntimeConfig) => {
    // Use gceConfig when PD feature is not enabled OR reduce PD size OR update an existing runtime with no PD attached.
    // TODO(RW-): 'Update an existing runtime with no PD attached' will be cleaned up post launch PD when all existing running Runtime shutdonw.
    if(!enablePD || pdSizeIncrease || (runtimeExists && !pdExists)) {
      return {
        gceConfig: {
          machineType: runtime.machine.name,
          diskSize: runtime.diskSize
        }
      };
    } else {
      return {
        gceWithPdConfig: {
          bootDiskSize: 50,
          machineType: runtime.machine.name,
          persistentDisk: {
            // When reducing PD size, passing empty name to backend then API will create a new PD
            name: pdSizeReduced ? '' : persistentDisk.name,
            size: runtime.diskSize,
            diskType: DiskType.Standard,
            labels: {}
          }
        }
      }
    }
  }
```